packages/lisk-transactions/src/utils/validate_multisignatures.ts

```diff
@@ -0,0 +1,54 @@
+import * as cryptography from '@liskhq/lisk-cryptography';
+import { Account, TransactionJSON } from '../transaction_types';
+
+export const validateMultisignatures = (
+	sender: Account,
+	transaction: TransactionJSON,
+	transactionHash: Buffer,
+): ReadonlyArray<string> => {
+	const minimumValidations = sender.multimin || 0;
+	const transactionSignatures = transaction.signatures as ReadonlyArray<string>;
+	const senderPublicKeys = sender.multisignatures || [];
+	// tslint:disable-next-line no-let
+	let validatedSignaturesCount = 0;
+	// tslint:disable-next-line no-let
+	let checkedPublicKeys: ReadonlyArray<string> = [];
```

Comment:
having this with reduce together seems a bit weird.
If it makes easier to understand safely, not using reduce is also an option.